omnigibson/prims/prim_base.py

```diff
@@ -71,26 +73,36 @@ def initialize(self):
         """
         assert (
             not self._initialized
-        ), f"Prim {self.name} at prim_path {self._prim_path} can only be initialized once! (It is already initialized)"
+        ), f"Prim {self.name} at prim_path {self.prim_path} can only be initialized once! (It is already initialized)"
         self._initialize()
 
-        # Cache state size
-        self._state_size = len(self.dump_state(serialized=True))
-
         self._initialized = True
 
-    def load(self):
+    def load(self, scene):
         """
         Load this prim into omniverse, and return loaded prim reference.
 
         Returns:
             Usd.Prim: Prim object loaded into the simulator
         """
-        if self._loaded:
-            raise ValueError(f"Cannot load prim {self.name} multiple times.")
+        # Load the prim if it doesn't exist yet.
+        assert (
+            not self._loaded
+        ), f"Prim {self.name} at prim_path {self.prim_path} can only be loaded once! (It is already loaded)"
+
+        # Assign the scene first.
+        self._scene = scene
+        self._scene_assigned = True
```

Comment:
normally i assume that `self._scene != None` is a sufficient check, but I suppose that a prim can be loaded with no scene?

Reply:
Sorry I'm a bit confused here: when you say `self._scene != None` is a sufficient check, what check are you referring to?

Reply:
re: Josiah, yes that's the reason.

Reply:
TODO: rename self._scene_assigned to something else that make sense